### PR TITLE
Migrate to IntelliJ Platform Gradle Plugin 2

### PR DESCRIPTION
**Pull request by mkondratek**
Closes https://github.com/sourcegraph/cody-issues/issues/240
Fixes https://linear.app/sourcegraph/issue/CODY-3713/upgrade-to-intellij-gradle-2x-plugin-to-debug-in-splitmode

## Test plan
1. Full QA
2. Green CI
3. runIde (and other gradle tasks) working

### CODE REVIEW

File: .intellijPlatform/localPlatformArtifacts/bundledPlugin-Git4Idea-2023.2+2.xml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<ivy-module version="2.0">
```
Comment (odisseus) [inline .intellijPlatform/localPlatformArtifacts/bundledPlugin-Git4Idea-2023.2+2.xml]: This directory should be added to `.gitignore`.

File: build.gradle.kts [hunk 2]
```diff
@@ -488,46 +514,25 @@ tasks {
...
-    // runtime plugins
-    // https://github.com/JetBrains/intellij-platform-gradle-plugin/issues/1489
-    // val platformRuntimePlugins = project.findProperty("platformRuntimePlugins")
-  }
-
-  runPluginVerifier {
-    ideVersions.set(versionsToValidate)
-    failureLevel.set(EnumSet.complementOf(skippedFailureLevels))
-  }
+    //    val platformRuntimeVersion = project.findProperty("platformRuntimeVersion")
```
Comment (pkukielka) [inline build.gradle.kts]: @mkondratek I commented this out but we need to adjust it to new API.
Looks like `version` and `type` are now properties of the task.
https://plugins.jetbrains.com/docs/intellij/tools-intellij-platform-gradle-plugin-testing-extension.html#intellijPlatformTesting

File: gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -1,6 +1,6 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-8.8-bin.zip
```
Comment (mkondratek) [inline gradle/wrapper/gradle-wrapper.properties:3]: 8.10.1 is available but it causes some troubles. Let's stick with 8.10 for now (and wait for 8.10.2).

https://github.com/gradle/gradle/issues/30497

File: build.gradle.kts [hunk 4]
```diff
@@ -488,46 +520,19 @@ tasks {
     }
   }
 
-  patchPluginXml {
-    sinceBuild = properties("pluginSinceBuild")
-    untilBuild = properties("pluginUntilBuild")
-  }
+  val customRunIde by
```
Comment (pkukielka) [inline build.gradle.kts:535]: Do we need separate task?
If we can set version on `runIde` then we can do the same, just with fallback to `platformVersion` if `platformRuntimeVersion` is not available.

**Comment by mkondratek** [comment]
```
Configuration file for j.u.l.LogManager does not exist: /Users/mkondratek/.gradle/caches/8.10/transforms/89d82208413f6e532a272dd6f85dc19a/transformed/ideaIC-2023.2-aarch64/test-log.properties
[    259]   WARN - o.j.n.NativeTrustedCertificates - Unable to get custom trusted certificates list from the operating system: 

There is an incompatible JNA native library installed on this system
Expected: 7.0.0
Found:    6.1.4
(at /Users/mkondratek/.gradle/caches/8.10/transforms/89d82208413f6e532a272dd6f85dc19a/transformed/ideaIC-2023.2-aarch64/lib/jna/aarch64/libjnidispatch.jnilib).
To resolve this issue you may do one of the following:
 - remove or uninstall the offending library
 - set the system property jna.nosys=true
 - set jna.boot.library.path to include the path to the version of the 
   jnidispatch library included with the JNA jar file you are using

java.lang.Error: 

There is an incompatible JNA native library installed on this system
Expected: 7.0.0
Found:    6.1.4
(at /Users/mkondratek/.gradle/caches/8.10/transforms/89d82208413f6e532a272dd6f85dc19a/transformed/ideaIC-2023.2-aarch64/lib/jna/aarch64/libjnidispatch.jnilib).
To resolve this issue you may do one of the following:
 - remove or uninstall the offending library
 - set the system property jna.nosys=true
 - set jna.boot.library.path to include the path to the version of the 
   jnidispatch library included with the JNA jar file you are using

	at com.sun.jna.Native.<clinit>(Native.java:230)
	at com.sun.jna.NativeLong.<clinit>(NativeLong.java:35)
	at org.jetbrains.nativecerts.NativeTrustedCertificates.getCustomOsSpecificTrustedCertificates(NativeTrustedCertificates.java:33)
	at com.intellij.util.net.ssl.OsCertificatesServiceImpl.getCustomOsSpecificTrustedCertificates(OsCertificatesServiceImpl.java:20)
	at com.intellij.util.net.ssl.ConfirmingTrustManager.getOperatingSystemTrustManager(ConfirmingTrustManager.java:93)
	at com.intellij.util.net.ssl.ConfirmingTrustManager.getSystemTrustManagers(ConfirmingTrustManager.java:60)
	at com.intellij.util.net.ssl.ConfirmingTrustManager.createForStorage(ConfirmingTrustManager.java:54)
	at com.intellij.util.net.ssl.CertificateManager$trustManager$2.invoke(CertificateManager.kt:64)
	at com.intellij.util.net.ssl.CertificateManager$trustManager$2.invoke(CertificateManager.kt:63)
	at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:74)
	at com.intellij.util.net.ssl.CertificateManager.getTrustManager(CertificateManager.kt:63)
	at com.intellij.util.net.ssl.CertificateManager.calcSslContext(CertificateManager.kt:285)
	at com.intellij.util.net.ssl.CertificateManager.access$calcSslContext(CertificateManager.kt:59)
	at com.intellij.util.net.ssl.CertificateManager$sslContext$2.invoke(CertificateManager.kt:83)
	at com.intellij.util.net.ssl.CertificateManager$sslContext$2.invoke(CertificateManager.kt:83)
	at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:74)
	at com.intellij.util.net.ssl.CertificateManager.getSslContext(CertificateManager.kt:83)
	at com.intellij.util.net.ssl.CertificateManager$1.invokeSuspend(CertificateManager.kt:92)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at kotlinx.coroutines.internal.LimitedDispatcher$Worker.run(LimitedDispatcher.kt:115)
	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:100)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:584)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:793)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:697)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:684)

[    261]   WARN - #c.i.u.n.s.ConfirmingTrustManager - Received an empty list of custom trusted root certificates from the system. Check log above for possible errors, enable debug logging in category 'org.jetbrains.nativecerts' for more information
[    692]   WARN - #c.i.j.JnaLoader - Unable to load JNA library (os=Mac OS X 14.5, jna.boot.library.path=/Users/mkondratek/.gradle/caches/8.10/transforms/89d82208413f6e532a272dd6f85dc19a/transformed/ideaIC-2023.2-aarch64/lib/jna/aarch64)
java.lang.NoClassDefFoundError: Could not initialize class com.sun.jna.Native
	at com.intellij.jna.JnaLoader.load(JnaLoader.java:19)
	at com.intellij.jna.JnaLoader.isLoaded(JnaLoader.java:35)
	at com.intellij.openapi.util.io.FileSystemUtil.readCaseSensitivityByNativeAPI(FileSystemUtil.java:390)
	at com.intellij.openapi.util.io.FileSystemUtil.readParentCaseSensitivity(FileSystemUtil.java:308)
	at com.intellij.openapi.vfs.newvfs.impl.VirtualDirectoryImpl.generateCaseSensitivityChangedEventForUnknownCase(VirtualDirectoryImpl.java:893)
	at com.intellij.openapi.vfs.newvfs.impl.VirtualDirectoryImpl.updateCaseSensitivityIfUnknown(VirtualDirectoryImpl.java:313)
	at com.intellij.openapi.vfs.newvfs.impl.VirtualDirectoryImpl.findChild(VirtualDirectoryImpl.java:82)
	at com.intellij.openapi.vfs.newvfs.impl.VirtualDirectoryImpl.refreshAndFindChild(VirtualDirectoryImpl.java:349)
	at com.intellij.openapi.vfs.newvfs.VfsImplUtil.refreshAndFindFileByPath(VfsImplUtil.java:117)
	at com.intellij.openapi.vfs.impl.local.LocalFileSystemBase.refreshAndFindFileByPath(LocalFileSystemBase.java:132)
	at com.intellij.testFramework.HeavyPlatformTestCase.synchronizeTempDirVfs(HeavyPlatformTestCase.java:165)
	at com.intellij.testFramework.LightPlatformTestCase.initProject(LightPlatformTestCase.java:180)
	at com.intellij.testFramework.LightPlatformTestCase.lambda$doSetup$3(LightPlatformTestCase.java:261)
	at com.intellij.openapi.application.impl.ApplicationImpl.invokeAndWait(ApplicationImpl.java:458)
	at com.intellij.openapi.application.impl.ApplicationImpl.invokeAndWait(ApplicationImpl.java:490)
	at com.intellij.testFramework.LightPlatformTestCase.doSetup(LightPlatformTestCase.java:252)
	at com.intellij.testFramework.fixtures.impl.LightIdeaTestFixtureImpl.setUp(LightIdeaTestFixtureImpl.java:41)
	at com.intellij.testFramework.fixtures.impl.CodeInsightTestFixtureImpl.lambda$setUp$38(CodeInsightTestFixtureImpl.java:1336)
	at com.intellij.testFramework.EdtTestUtil.runInEdtAndWait(EdtTestUtil.java:33)
	at com.intellij.testFramework.fixtures.impl.CodeInsightTestFixtureImpl.setUp(CodeInsightTestFixtureImpl.java:1335)
	at com.intellij.testFramework.fixtures.BasePlatformTestCase.setUp(BasePlatformTestCase.java:48)
	at com.sourcegraph.cody.agent.protocol.PositionTest.setUp(PositionTest.kt:22)
	at com.intellij.testFramework.UsefulTestCase.invokeSetUp(UsefulTestCase.java:426)
	at com.intellij.testFramework.UsefulTestCase.defaultRunBare(UsefulTestCase.java:418)
	at com.intellij.testFramework.UsefulTestCase.lambda$runBare$12(UsefulTestCase.java:487)
	at com.intellij.testFramework.EdtTestUtil.lambda$runInEdtAndWait$1(EdtTestUtil.java:40)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:308)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:789)
	at java.desktop/java.awt.EventQueue$3.run(EventQueue.java:740)
	at java.desktop/java.awt.EventQueue$3.run(EventQueue.java:734)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:759)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.kt:322)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:207)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:128)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:117)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:105)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:92)
Caused by: java.lang.ExceptionInInitializerError: Exception java.lang.Error: 

There is an incompatible JNA native library installed on this system
Expected: 7.0.0
Found:    6.1.4
(at /Users/mkondratek/.gradle/caches/8.10/transforms/89d82208413f6e532a272dd6f85dc19a/transformed/ideaIC-2023.2-aarch64/lib/jna/aarch64/libjnidispatch.jnilib).
To resolve this issue you may do one of the following:
 - remove or uninstall the offending library
 - set the system property jna.nosys=true
 - set jna.boot.library.path to include the path to the version of the 
   jnidispatch library included with the JNA jar file you are using
 [in thread "DefaultDispatcher-worker-9 @com.intellij.openapi.application.impl.ApplicationImpl#52"]
	at com.sun.jna.Native.<clinit>(Native.java:230)
	at com.sun.jna.NativeLong.<clinit>(NativeLong.java:35)
	at org.jetbrains.nativecerts.NativeTrustedCertificates.getCustomOsSpecificTrustedCertificates(NativeTrustedCertificates.java:33)
	at com.intellij.util.net.ssl.OsCertificatesServiceImpl.getCustomOsSpecificTrustedCertificates(OsCertificatesServiceImpl.java:20)
	at com.intellij.util.net.ssl.ConfirmingTrustManager.getOperatingSystemTrustManager(ConfirmingTrustManager.java:93)
	at com.intellij.util.net.ssl.ConfirmingTrustManager.getSystemTrustManagers(ConfirmingTrustManager.java:60)
	at com.intellij.util.net.ssl.ConfirmingTrustManager.createForStorage(ConfirmingTrustManager.java:54)
	at com.intellij.util.net.ssl.CertificateManager$trustManager$2.invoke(CertificateManager.kt:64)
	at com.intellij.util.net.ssl.CertificateManager$trustManager$2.invoke(CertificateManager.kt:63)
	at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:74)
	at com.intellij.util.net.ssl.CertificateManager.getTrustManager(CertificateManager.kt:63)
	at com.intellij.util.net.ssl.CertificateManager.calcSslContext(CertificateManager.kt:285)
	at com.intellij.util.net.ssl.CertificateManager.access$calcSslContext(CertificateManager.kt:59)
	at com.intellij.util.net.ssl.CertificateManager$sslContext$2.invoke(CertificateManager.kt:83)
	at com.intellij.util.net.ssl.CertificateManager$sslContext$2.invoke(CertificateManager.kt:83)
	at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:74)
	at com.intellij.util.net.ssl.CertificateManager.getSslContext(CertificateManager.kt:83)
	at com.intellij.util.net.ssl.CertificateManager$1.invokeSuspend(CertificateManager.kt:92)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at kotlinx.coroutines.internal.LimitedDispatcher$Worker.run(LimitedDispatcher.kt:115)
	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:100)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:584)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:793)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:697)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:684)

JNA library is not available
java.lang.AssertionError: JNA library is not available
	at com.intellij.ui.mac.foundation.Foundation.<clinit>(Foundation.java:30)
	at com.intellij.ui.mac.foundation.Foundation$NSAutoreleasePool.<init>(Foundation.java:480)
	at com.intellij.ui.components.MacScrollBarUI.callMac(MacScrollBarUI.java:181)
	at com.intellij.ui.components.MacScrollBarUI$Native.<init>(MacScrollBarUI.java:281)
	at com.intellij.ui.components.MacScrollBarUI$Style$1.<init>(MacScrollBarUI.java:232)
	at com.intellij.ui.components.MacScrollBarUI$Style.<clinit>(MacScrollBarUI.java:232)
	at com.intellij.ui.components.MacScrollBarUI.installUI(MacScrollBarUI.java:86)
	at java.desktop/javax.swing.JComponent.setUI(JComponent.java:733)
	at java.desktop/javax.swing.JScrollBar.setUI(JScrollBar.java:218)
	at com.intellij.openapi.editor.impl.EditorImpl$MyScrollBar.setUI(EditorImpl.java:3187)
	at com.intellij.ui.components.JBScrollBar.updateUI(JBScrollBar.java:87)
	at java.desktop/javax.swing.JScrollBar.<init>(JScrollBar.java:173)
	at com.intellij.ui.components.JBScrollBar.<init>(JBScrollBar.java:65)
	at com.intellij.ui.components.JBScrollBar.<init>(JBScrollBar.java:61)
	at com.intellij.openapi.editor.impl.EditorImpl$OpaqueAwareScrollBar.<init>(EditorImpl.java:3143)
	at com.intellij.openapi.editor.impl.EditorImpl$MyScrollBar.<init>(EditorImpl.java:3176)
	at com.intellij.openapi.editor.impl.EditorImpl$MyScrollPane.createVerticalScrollBar(EditorImpl.java:5266)
	at java.desktop/javax.swing.JScrollPane.<init>(JScrollPane.java:301)
	at java.desktop/javax.swing.JScrollPane.<init>(JScrollPane.java:353)
	at com.intellij.ui.components.JBScrollPane.<init>(JBScrollPane.java:83)
	at com.intellij.openapi.editor.impl.EditorImpl$MyScrollPane.<init>(EditorImpl.java:5203)
	at com.intellij.openapi.editor.impl.EditorImpl.<init>(EditorImpl.java:368)
	at com.intellij.openapi.editor.impl.EditorFactoryImpl.createEditor(EditorFactoryImpl.java:210)
	at com.intellij.openapi.editor.impl.EditorFactoryImpl.createEditor(EditorFactoryImpl.java:165)
	at com.intellij.openapi.fileEditor.impl.TestEditorManagerImpl.doOpenTextEditor(TestEditorManagerImpl.kt:456)
	at com.intellij.openapi.fileEditor.impl.TestEditorManagerImpl.openFileImpl3(TestEditorManagerImpl.kt:147)
	at com.intellij.openapi.fileEditor.impl.TestEditorManagerImpl.openFileInCommand$lambda$6(TestEditorManagerImpl.kt:436)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:219)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:174)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:164)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:150)
	at com.intellij.openapi.fileEditor.impl.TestEditorManagerImpl.openFileInCommand(TestEditorManagerImpl.kt:435)
	at com.intellij.openapi.fileEditor.impl.TestEditorManagerImpl.openTextEditor(TestEditorManagerImpl.kt:425)
	at com.intellij.testFramework.fixtures.impl.CodeInsightTestFixtureImpl.createEditor(CodeInsightTestFixtureImpl.java:1649)
	at com.intellij.testFramework.fixtures.impl.CodeInsightTestFixtureImpl.openFileInEditor(CodeInsightTestFixtureImpl.java:651)
	at com.sourcegraph.cody.agent.protocol.PositionTest.setUp(PositionTest.kt:23)
	at com.intellij.testFramework.UsefulTestCase.invokeSetUp(UsefulTestCase.java:426)
	at com.intellij.testFramework.UsefulTestCase.defaultRunBare(UsefulTestCase.java:418)
	at com.intellij.testFramework.UsefulTestCase.lambda$runBare$12(UsefulTestCase.java:487)
	at com.intellij.testFramework.EdtTestUtil.lambda$runInEdtAndWait$1(EdtTestUtil.java:40)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:308)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:789)
	at java.desktop/java.awt.EventQueue$3.run(EventQueue.java:740)
	at java.desktop/java.awt.EventQueue$3.run(EventQueue.java:734)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:759)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.kt:322)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:207)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:128)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:117)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:105)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:92)

```

Locally I see this error for some tests.

File: build.gradle.kts [hunk 5]
```diff
@@ -53,42 +56,84 @@ plugins {
...
+    create(platformType, platformVersion)
+    bundledPlugins(
+        properties("platformPlugins")
+            .orEmpty()
+            .split(',')
+            .map(String::trim)
+            .filter(String::isNotEmpty))
+    instrumentationTools()
+    pluginVerifier()
+    testFramework(TestFrameworkType.Platform)
```
Comment (mkondratek) [inline build.gradle.kts:129]: https://plugins.jetbrains.com/docs/intellij/tools-intellij-platform-gradle-plugin-types.html#TestFrameworkType

File: build.gradle.kts (hunk 4 above)
Comment (mkondratek) [reply build.gradle.kts:535]: sadly, it does not work yet

**Review by pkukielka — APPROVED** [review]
LGTM

**Pull request merged**